### PR TITLE
Fix: Flush DB to disk more often

### PR DESCRIPTION
**Pull request by jorgemmsilva**
In order to prevent what happened last week with the broken access node that was forcefully stopped and couldn't recuperate, this PR makes some changes so that :
 - Chain registry changes are always persisted on disk
 - Chain blocks are written flushed to disk every time a new "state transition" happens

### CODE REVIEW

File: packages/vm/processors/cache.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 
 // Cache stores all initialized VMProcessor instances used by a single chain
 type Cache struct {
-	*sync.Mutex
+	Mutex      *sync.Mutex
```
Comment (dessaya) [inline packages/vm/processors/cache.go]: Can it be private?

File: packages/database/registrykvstore/registrykvstore.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+func (s *RegistryKVStore) Get(key kvstore.Key) (value kvstore.Value, err error) {
+	return s.store.Get(key)
+}
+
+func (s *RegistryKVStore) Set(key kvstore.Key, value kvstore.Value) error {
+	err := s.store.Set(key, value)
+	if err != nil {
+		return err
+	}
+	return s.store.Flush()
```
Comment (dessaya) [inline packages/database/registrykvstore/registrykvstore.go:49]: Could this result in a negative performance impact? I don't know how hive / rocksdb work but in general the DB should be "intelligent", and have its own write-ahead log, and know what's the best time to flush.

**Review by dessaya — APPROVED** [review]

File: packages/database/registrykvstore/registrykvstore.go (hunk 2 above)
Comment (jorgemmsilva) [reply packages/database/registrykvstore/registrykvstore.go:49]: writes to the chain-registry are pretty sparse. It should impact performance when that happens, but I don't think its meaningful, and it keeps the node "safer" from forced shutdowns

File: packages/vm/processors/cache.go (hunk 1 above)
Comment (jorgemmsilva) [reply packages/vm/processors/cache.go]: good point, updated

**Pull request merged**